source/common/config/well_known_names.cc

```diff
@@ -24,97 +24,95 @@ TagNameValues::TagNameValues() {
+  addRe2(MONGO_CMD, R"(^mongo\.[\w-]+\.cmd\.(([\w-]+)\.))", ".cmd.");
 
-  // cluster.[<route_target_cluster>.]grpc.[<grpc_service>.](<grpc_method>.)<base_stat>
-  addRegex(GRPC_BRIDGE_METHOD, R"(^cluster(?=\.).*?\.grpc(?=\.).*\.((.*?)\.)\w+?$)", ".grpc.");
+  // cluster.[<route_target_cluster>.]grpc.<grpc_service>.(<grpc_method>.)*
+  addRe2(GRPC_BRIDGE_METHOD, R"(^cluster\.[\w-]+\.grpc\.[\w-]+\.(([\w-]+)\.))", ".grpc.");
 
-  // http.[<stat_prefix>.]user_agent.(<user_agent>.)<base_stat>
-  addRegex(HTTP_USER_AGENT, R"(^http(?=\.).*?\.user_agent\.((.*?)\.)\w+?$)", ".user_agent.");
+  // http.[<stat_prefix>.]user_agent.(<user_agent>.)*
+  addRe2(HTTP_USER_AGENT, R"(^http\.[\w-]+\.user_agent\.(([\w-]+)\.))", ".user_agent.");
 
-  // vhost.[<virtual host name>.]vcluster.(<virtual_cluster_name>.)<base_stat>
-  addRegex(VIRTUAL_CLUSTER, R"(^vhost(?=\.).*?\.vcluster\.((.*?)\.)\w+?$)", ".vcluster.");
+  // vhost.[<virtual host name>.]vcluster.(<virtual_cluster_name>.)*
+  addRe2(VIRTUAL_CLUSTER, R"(^vhost\.[\w-]+\.vcluster\.(([\w-]+)\.))", ".vcluster.");
 
-  // http.[<stat_prefix>.]fault.(<downstream_cluster>.)<base_stat>
-  addRegex(FAULT_DOWNSTREAM_CLUSTER, R"(^http(?=\.).*?\.fault\.((.*?)\.)\w+?$)", ".fault.");
+  // http.[<stat_prefix>.]fault.(<downstream_cluster>.)*
+  addRe2(FAULT_DOWNSTREAM_CLUSTER, R"(^http\.[\w-]+\.fault\.(([\w-]+)\.))", ".fault.");
 
   // listener.[<address>.]ssl.cipher.(<cipher>)
-  addRegex(SSL_CIPHER, R"(^listener(?=\.).*?\.ssl\.cipher(\.(.*?))$)");
+  addRe2(SSL_CIPHER, R"(^listener\..*?\.ssl\.cipher(\.([0-9A-Za-z_-]+))$)");
 
   // cluster.[<cluster_name>.]ssl.ciphers.(<cipher>)
-  addRegex(SSL_CIPHER_SUITE, R"(^cluster(?=\.).*?\.ssl\.ciphers(\.(.*?))$)", ".ssl.ciphers.");
+  addRe2(SSL_CIPHER_SUITE, R"(^cluster\.[\w-]+\.ssl\.ciphers(\.([0-9A-Za-z_-]+))$)",
+         ".ssl.ciphers.");
 
   // cluster.[<route_target_cluster>.]grpc.(<grpc_service>.)*
-  addRegex(GRPC_BRIDGE_SERVICE, R"(^cluster(?=\.).*?\.grpc\.((.*?)\.))", ".grpc.");
+  addRe2(GRPC_BRIDGE_SERVICE, R"(^cluster\.[\w-]+\.grpc\.(([\w-]+)\.))", ".grpc.");
 
   // tcp.(<stat_prefix>.)<base_stat>
-  addRegex(TCP_PREFIX, R"(^tcp\.((.*?)\.)\w+?$)");
+  addRe2(TCP_PREFIX, R"(^tcp\.(([\w-]+)\.))");
 
   // udp.(<stat_prefix>.)<base_stat>
-  addRegex(UDP_PREFIX, R"(^udp\.((.*?)\.)\w+?$)");
+  addRe2(UDP_PREFIX, R"(^udp\.(([\w-]+)\.))");
 
-  // auth.clientssl.(<stat_prefix>.)<base_stat>
-  addRegex(CLIENTSSL_PREFIX, R"(^auth\.clientssl\.((.*?)\.)\w+?$)");
+  // auth.clientssl.(<stat_prefix>.)*
+  addRe2(CLIENTSSL_PREFIX, R"(^auth\.clientssl\.(([\w-]+)\.))");
 
-  // ratelimit.(<stat_prefix>.)<base_stat>
-  addRegex(RATELIMIT_PREFIX, R"(^ratelimit\.((.*?)\.)\w+?$)");
+  // ratelimit.(<stat_prefix>.)*
+  addRe2(RATELIMIT_PREFIX, R"(^ratelimit\.(([\w-]+)\.))");
 
   // cluster.(<cluster_name>.)*
-  addRe2(CLUSTER_NAME, "^cluster\\.(([^\\.]+)\\.).*");
+  addRe2(CLUSTER_NAME, R"(^cluster\.(([\w-]+)\.))");
```

Comment:
I have a question on the use of `\w` here -- does this possibly reject expressions that the original one did not? I think the answer is "yes". E.g. punctuation other than "." or "-" that might be expected to work. I think cluster-names are in control of the user, so this might be a visible change in behavior.

Is there a speed difference for using `\w-` vs `^\.` ?

Reply:
`^\.` seems to perform better. Perhaps this is a negated comparison with a single symbol vs comparisons with multiple symbols. Replaced with `^\.` everywhere.